pkg/operator/reconciler/workflow/reconciler.go

```diff
@@ -149,6 +146,15 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (result ct
 		return ctrl.Result{}, nil
 	}
 
+	// FIXME Refactor...
+	if len(wrd.Workflow.Object.Spec.Steps) == 0 {
```

Comment:
You could probably just make this check `else if pr == nil { ... }` which would generally indicate that we don't need a `PipelineRun` for this run for whatever reason and it would feel clean the way you have it implemented otherwise.